packages/babel-preset-typescript/src/index.js

```diff
@@ -21,7 +21,7 @@ export default declare(
     if (typeof allExtensions !== "undefined" || typeof isTSX !== "undefined") {
       throw new Error(
         "The .allExtensions and .isTSX options have been removed.\n" +
-          "If you want to disable JSX detection based on the file extension, " +
+          "If you want to disable file extension-based JSX detection, " +
```

Comment:
English needs parentheses, here the fact that ` ` between "file" and "extension" has a higher precedence than `-` feels... weird :stuck_out_tongue: